bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -1041,11 +1041,14 @@ func (r *KubeadmConfigReconciler) ensureBootstrapSecretOwnersRef(ctx context.Con
 	if err != nil {
 		return errors.Wrapf(err, "failed to add KubeadmConfig %s as ownerReference to bootstrap Secret %s", scope.ConfigOwner.GetName(), secret.GetName())
 	}
+	if controller := metav1.GetControllerOf(secret); controller != nil {
+		util.RemoveOwnerRef(secret.OwnerReferences, *controller)
```

Comment:
does this actually modify secret.OwnerReferences or does it return a new slice with the controller removed?

(if I'm right, do we have this problem somewhere else as well?)